core/src/main/java/org/elasticsearch/index/query/GeoBoundingBoxQueryBuilder.java

```diff
@@ -98,95 +160,344 @@ public GeoBoundingBoxQueryBuilder bottomRight(String geohash) {
+                validationException = addValidationError("right longitude is invalid number: " + bottomRight.getLon(), validationException);
+            }
+
+            if (validationException == null) {
+                // all corners are valid after above checks - make sure they are in the right relation
+                if (topLeft.getLat() < bottomRight.getLat()) {
+                    validationException = addValidationError("top is below bottom corner: " +
+                            topLeft.getLat() + " vs. " + bottomRight.getLat(), validationException);
+                }
+
+                // we do not check longitudes as the query generation code can deal with flipped left/right values
+            }
         }
 
-        builder.startObject(NAME);
+        if (Strings.isEmpty(fieldName)) {
+            validationException = addValidationError("field name must be non-null and non-empty", validationException);
+        }
+        return validationException;
+    }
 
-        builder.startObject(name);
-        builder.array(TOP_LEFT, box[LEFT], box[TOP]);
-        builder.array(BOTTOM_RIGHT, box[RIGHT], box[BOTTOM]);
-        builder.endObject();
+    QueryValidationException validate(boolean indexCreatedBeforeV2_0) {
+        if (ignoreMalformed || indexCreatedBeforeV2_0) {
+            return null;
+        }
+
+        QueryValidationException validationException = null;
+        // For everything post 2.0 validate latitude and longitude unless validation was explicitly turned off
+        if (GeoUtils.isValidLatitude(topLeft.getLat()) == false) {
+            validationException = addValidationError("top latitude is invalid: " + topLeft.getLat(),
+                    validationException);
+        }
+        if (GeoUtils.isValidLongitude(topLeft.getLon()) == false) {
+            validationException = addValidationError("left longitude is invalid: " + topLeft.getLon(),
+                    validationException);
+        }
+        if (GeoUtils.isValidLatitude(bottomRight.getLat()) == false) {
+            validationException = addValidationError("bottom latitude is invalid: " + bottomRight.getLat(),
+                    validationException);
+        }
+        if (GeoUtils.isValidLongitude(bottomRight.getLon()) == false) {
+            validationException = addValidationError("right longitude is invalid: " + bottomRight.getLon(),
+                    validationException);
+        }
+        return validationException;
+    }
 
-        if (type != null) {
-            builder.field("type", type);
+    @Override
+    public Query doToQuery(QueryShardContext context) {
+        QueryValidationException exception = validate(context.indexVersionCreated().before(Version.V_2_0_0));
+        if (exception != null) {
+            throw new QueryShardException(context, "couldn't validate latitude/ longitude values", exception);
+        }
+
+        if (context.mapperService().smartNameFieldType(fieldName, context.getTypes()) == null){
```

Comment:
+1 to making the change in master. Will post a PR.

If I remove the if I get a [ih] QueryShardException[failed to find geo_point field [mapped_geo]] for the test cases where we have no mapping for geo_points. The behaviour for this case is undefined - according to https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-geo-bounding-box-filter.html#_geo_point_type the mapping has to be there for geo bounding box queries to work.